### ISSUE TITLE
Поддержка корпоративных пользователей

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vacuum-im/vacuum-im#490]
```
Здесь будет список задач и обсуждение 
доработок клиента для удобного 
использования в корпоративных сетях.
```

Original issue reported on code.google.com by `potapov.s.a` on 24 Aug 2011 at 3:55
- Blocked on: #64, #277, #462, #505

**Comment by StreamThreader** [comment]
Для тихой установки MSI файла политикой MS AD надо какие-то параметры отредактирлвать с помощью Orca ?
Если да, то подскажите каие параметры необходимо поправить, что бы реализовать рекомендуемые аргументы установки (msiexec /i vacuum-im_~~AllUsers.msi /quiet /qn /norestart /log "%UserProfile%\vacuum-deploy-install.log").
Я до этого не сталкивался с разворачиванием ПО с помощью групповых политик, и не мог даже предположить, что там нет возможности задать ключи установки.

**Comment by krab** [comment]
AllUsers.msi устанавливается и в тихом режиме ничего править не нужно,
возможно нужно будет настроить uac на выполнение программ без подтверждения
там же в gpo и разрешить установку из не доверенных источников.
21 янв. 2016 г. 5:16 PM пользователь "Oleg A. Deordiev" <
notifications@github.com> написал:

**Comment by StreamThreader** [comment]
В настройках групповой политики, можно указать файл MSI, но я не могу найти где там можно указать ключи запуска. У меня Win2k3 R2.

**Comment by krab** [comment]
Никакие ключи не нужны. :D
21 янв. 2016 г. 7:37 PM пользователь "Oleg A. Deordiev" <
notifications@github.com> написал:

**Comment by dreamer2** [comment]
не могу запустить ntlm подключение, не понимаю, что писать в логин пользователю, с пустым не работает
что не пробую ругается аккаунт неверно настроен или спрашивает пароль для учетки

плагин установлен, опция "разрешить ntlm" появилась

сервер openfire с патчем, миранде достаточно галочки "доменный логин"

**Comment by Lion34** [comment]
В логин надо писать JID пользователя, пароль можно оставить пустым. Если не поможет, пришлите XML лог подключения.

**Comment by dreamer2** [comment]
с логином значит пробовал правильно

такой лог?
[verr.txt](https://github.com/Vacuum-IM/vacuum-im/files/130232/verr.txt)

**Comment by Lion34** [comment]
Очень странный лог, без NTLM нормально подключается? Выложи еще лог самого приложения, он должен тут быть c:\Users<user>\AppData\Roaming\JRuDevels\VacuumIM\logs\

**Comment by dreamer2** [comment]
да, кстати, дело в том что он открывает окно ввода пароля и судя по логу даже не пытается войти автоматически

если пароль вводить руками далее будет
`<auth xmlns="urn:ietf:params:xml:ns:xmpp-sasl" mechanism="PLAIN">AGRvZG9ub ......`

в случае миранды автоматом далее идет
`
<auth xmlns="urn:ietf:params:xml:ns:xmpp-sasl" mechanism="NTLM">TlRMTVNTUAABAAAAB7IIog4ADgAxAAAACQAJACgAAAAGA4AlAAAAD0tPTS1JTkYtREFETUlOSVNUUkFUSU9O</auth>
­<challenge xmlns="urn:ietf:params:xml:ns:xmpp-sasl">TlRMTV ..........`

я почему и спрашиваю, если судя по логу он даже не пытается использовать ntlm, может быть у меня еще что-то не настроено?

лог [2016-02-16T08-50-03.617+03-00.txt](https://github.com/Vacuum-IM/vacuum-im/files/131953/2016-02-16T08-50-03.617.03-00.txt)

**Comment by Lion34** [comment]
Судя по логам, либо отключена настройка, либо не инициализируется NTLM в винде, если второе, то есть смысл попробовать на другой машине или посмотреть политики в этом направлении, может запустить от админа для проверки.

**Comment by dreamer2** [comment]
отключена настройка котороая?
в плагинах есть включенные ntlm авторизация и sasl
в аккаунте "использовать параметры системного пользователя для авторизации" стоит

по винде напоминаю что миранда тут же работает
хотя от имени администратора конечно проверил, не помогло

есть смысл сетевой обмен сдампить? или если в логе нет попыток послать какой-либо auth значит клиент даже не слал ничего, проблема в настройках или определении возможностей системы?

windows 8.1 x64

**Comment by Lion34** [comment]
В NTLM плагине найдена ошибка, из-за которой он мог не работать, попробуй новую версию (скоро соберу).

**Comment by Lion34** [comment]
Обновил NTLM плагин в том числе и для версии 1.2

**Comment by Lion34** [comment]
В последнее время резко возросло количество ошибок при загрузке настроек "Failed to load options from file content: error occurred while parsing element", не могу понять в чем проблема, почему не получается прочитать файл, который сам же сохраняю. Посмотрите, пожалуйста, нет ли у ваших пользователей файла options.xml.fail в папке c:\Users<user>\AppData\Roaming\JRuDevels\VacuumIM\profiles\Default?

**Comment by dreamer2** [comment]
есть такой файл

все последние сборки этого года не работают с ntlm, с чем это связано? #904

**Comment by Lion34** [comment]
Проблемы с NTLM поправил.

**Comment by Fafofu** [comment]
Добрый день. А "Vacuum" поддерживает MST-файл ответов? Что-то я "Orca" запустил, и не вижу чтоб были доступны какие либо настройки :(

**Comment by krab** [comment]
Да. Но там нечего менять, WIX Toolset все практически прячет.
Каталог установки можно поменять через: 
`msiexec /q APPLICATIONFOLDER=C:\VacuumIM setup_file.msi`

Естественно на системные каталоги только для vacuum-im_1.2.5.20150608_win32-msvc2010_qt487_GPO.msi

**Comment by Fafofu** [comment]
Хотелось бы, чтоб можно было заранее задать, адрес сервера, галочки в настройках расставить как надо, указать в качестве расположение переменную с именем компьютера. Чтоб развернуть быстро в организации...

**Comment by Lion34** [comment]
Для этого нужно использовать свой шаблон настроек, об этом написано выше в этой теме.

**Comment by Fafofu** [comment]
Да тут столько понаписано, мне все 1000 сообщений перечитывать!? Вынесете в FAQ раз есть такая возможность. Или мне нужно будет им ещё ручками потом раскладывать файл с настройками?

**Comment by jbr2017** [comment]
Хочу создать файл с готовыми настройками, но для разных пользователей, чтобы постоянно не настраивать профиль.
Файлы, про которые здесь писали в папках Vacuum не нашёл.
default-options.xml 
static-options.xml - настройки применяемый при 
каждом открытии профиля
initial-options.xml - настройки применяемые один раз 
при создании профиля

Как создать такой файл?
